### PR TITLE
feat: custom-cursor

### PR DESCRIPTION
**Pull request by AnkushS27**

### CODE REVIEW

File: src/app/page.tsx [hunk 1]
```diff
@@ -24,11 +34,39 @@ export default function Home() {
     }
   }, [progress]);
 
+  useEffect(() => {
+    (async () => {
+      const lenis = new Lenis({
+        lerp: 0.1,
+        wheelMultiplier: 0.6,
+        touchMultiplier: 0.6,
+      });
```
Comment (manavbabber) [inline src/app/page.tsx:43]: should have proper cleanup

File: src/components/customCursor/customCursor.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      gsap.set('.cursor', { display: 'block' });
+      gsap.to('.cursorFollower', {
+        scale: 1,
+      });
+    };
+
+    document.querySelectorAll('.hovering').forEach((el) => {
+      el.addEventListener('mouseenter', mouseEnter);
+      el.addEventListener('mouseleave', mouseLeave);
+    });
```
Comment (manavbabber) [inline src/components/customCursor/customCursor.tsx:43]: event listeners should be removed before the next rerender, in the cleanup function

File: src/components/section/servicesSection.tsx [hunk 3]
```diff
@@ -3,10 +3,25 @@ import Image from 'next/image';
 import { Akzidenzgroteskbe, IBMPlexMono } from '@/fonts/fonts';
 
 export default function ServicesSection() {
+  const serviceItem = [
```
Comment (manavbabber) [inline src/components/section/servicesSection.tsx:6]: Shift this data to a separate file.

**Review by manavbabber — CHANGES_REQUESTED** [review]

File: src/app/page.tsx (hunk 1 above)
Comment (AnkushS27) [reply src/app/page.tsx:43]: done

File: src/components/section/servicesSection.tsx (hunk 3 above)
Comment (AnkushS27) [reply src/components/section/servicesSection.tsx:6]: done

File: src/components/customCursor/customCursor.tsx (hunk 2 above)
Comment (AnkushS27) [reply src/components/customCursor/customCursor.tsx:43]: done

**Review by rahulghangas — APPROVED** [review]

**Pull request merged**